contracts/Pool.sol

```diff
@@ -14,260 +15,368 @@ import "./IBorrowersRegistry.sol";
  * Rates are compounded every second and getters always return the current deposit and borrowing balance.
  * The interest rates calculation is delegated to the external calculator contract.
  */
-contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable {
+contract Pool is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC20 {
 
-    uint256 public constant MAX_POOL_UTILISATION = 0.95 ether;
+  uint256 public constant MAX_POOL_UTILISATION = 0.95 ether;
 
-    mapping(address => uint256) public deposits;
-    uint256 public totalDeposited;
+  mapping(address => mapping(address => uint256)) private _allowed;
+  mapping(address => uint256) private _deposited;
 
-    mapping(address => uint256) public borrowed;
-    uint256 public totalBorrowed;
+  mapping(address => uint256) public borrowed;
 
-    IRatesCalculator ratesCalculator;
-    IBorrowersRegistry borrowersRegistry;
+  IRatesCalculator private _ratesCalculator;
+  IBorrowersRegistry private _borrowersRegistry;
 
-    CompoundingIndex depositIndex;
-    CompoundingIndex borrowIndex;
+  CompoundingIndex depositIndex = new CompoundingIndex();
+  CompoundingIndex borrowIndex = new CompoundingIndex();
 
 
-    function initialize(
-      IRatesCalculator _ratesCalculator,
-      IBorrowersRegistry _borrowersRegistry,
-      CompoundingIndex _depositIndex,
-      CompoundingIndex _borrowIndex
+  function initialize(
+      IRatesCalculator ratesCalculator_,
+      IBorrowersRegistry borrowersRegistry_,
```

Comment:
How is it that the second parameter is `IBorrowersRegistry borrowersRegistry_` but in the `6_loans_factory.js` we pass factory (SmartLoansFactory) instance as the second parameter?

Reply:
Because in our case the `SmartLoansFactory` contract `is` an `IBorrowersRegistry` :)